lib/s3routes/routes/routeDELETE.ts

```diff
@@ -54,11 +54,14 @@ export default function routeDELETE(
         api.callApiMethod('objectDelete', request, response, log,
             (err, corsHeaders) => {
                 /*
-              * Since AWS expects a 204 regardless of the existence of
-              the object, the errors NoSuchKey and NoSuchVersion should not
-              * be sent back as a response.
-              */
-                if (err && !err.is.NoSuchKey && !err.is.NoSuchVersion) {
+                 * Since AWS expects a 204 regardless of the existence of
+                 * the object, the errors NoSuchKey and NoSuchVersion should not
+                 * be sent back as a response.
+                 */
+                if (err && (
+                    !(err instanceof ArsenalError) || 
+                    (err.is?.NoSuchKey === false && err.is?.NoSuchVersion === false)
```

Comment:
nit: the `?` is not needed anymore in that case, since we are garanteed to have an `ArsenalError`.

(and best to use bool equivalence than strict equality to false, IMO...)

```suggestion
                    (!err.is.NoSuchKey && !err.is.NoSuchVersion)
```

Reply:
done